packages/common-server/src/git.ts

```diff
@@ -140,6 +141,63 @@ export class GitUtils {
     return path.basename(url, ".git");
   }
 
+  /** Find the dependency path for a vault given the remote url. You can use
+   * this even if the vault has no remote.
+   *
+   * This is the relative path within the dependencies folder, like
+   * `github.com/dendronhq/dendron-site`. For more details see the
+   * [[Self Contained Vaults RFC|dendron://dendron.docs/rfc.42-self-contained-vaults]]
+   */
+  public static remoteUrlToDependencyPath({
+    vaultName,
+    url,
+  }: {
+    vaultName: string;
+    url?: string;
+  }): string {
+    // If no remote URL exists, then it's a local vault. We keep these in a
+    // local-only folder.
+    if (url === undefined) return FOLDERS.LOCAL_DEPENDENCY;
+    // Check if it matches any web URLs like
+    // https://github.com/dendronhq/dendron-site.git This may also look like
+    // http://example.com:8000/dendronhq/dendron-site.git, we skip the port
+    const webMatch =
+      /^(https?:\/\/)(?<domain>[^/:]+)(:[0-9]+)?\/(?<path>.+?)(\.git)?$/.exec(
```

Comment:
I suggested this during a pairing session, but I am now realizing that js doesn't natively support verbose regex 😅 

I was imagining something like what python has ([re.VERBOSE](https://docs.python.org/3/library/re.html#re.VERBOSE))

Looks like there are a few packages that implement this but not sure if it will be worth for us to add another dependency for this. We might get away with documenting them in a way we would do if verbose flag was available.

Reply:
It's still kinda possible by using strings and disabling prettier formatting for that block. I'll go ahead and make that change.